pkg/controller/networkpolicy/validate.go

```diff
@@ -327,40 +330,64 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(curObj, oldObj interface{}
 		for _, val := range v.antreaPolicyValidators {
 			reason, allowed = val.deleteValidate(oldObj, userInfo)
 			if !allowed {
-				return reason, allowed
+				return warnings, reason, allowed
 			}
 		}
 	}
-	return reason, allowed
+	return warnings, reason, allowed
 }
 
-func (v *NetworkPolicyValidator) validateAdminNetworkPolicy(curObj, oldObj interface{}, op admv1.Operation, userInfo authenticationv1.UserInfo) (string, bool) {
+func (v *NetworkPolicyValidator) validateAdminNetworkPolicy(curObj, oldObj interface{}, op admv1.Operation, userInfo authenticationv1.UserInfo) ([]string, string, bool) {
 	allowed := true
 	reason := ""
+	var warnings []string
 	switch op {
 	case admv1.Create:
 		for _, val := range v.adminNPValidators {
-			reason, allowed = val.createValidate(curObj, userInfo)
+			warnings, reason, allowed = val.createValidate(curObj, userInfo)
 			if !allowed {
-				return reason, allowed
+				return warnings, reason, allowed
 			}
 		}
 	case admv1.Update:
 		for _, val := range v.adminNPValidators {
-			reason, allowed = val.updateValidate(curObj, oldObj, userInfo)
+			warnings, reason, allowed = val.updateValidate(curObj, oldObj, userInfo)
 			if !allowed {
-				return reason, allowed
+				return warnings, reason, allowed
 			}
 		}
 	case admv1.Delete:
 		for _, val := range v.adminNPValidators {
 			reason, allowed = val.deleteValidate(oldObj, userInfo)
 			if !allowed {
-				return reason, allowed
+				return warnings, reason, allowed
 			}
 		}
 	}
-	return reason, allowed
+	return warnings, reason, allowed
+}
+
+func (v *antreaPolicyValidator) checkLogLabel(specAppliedTo []crdv1beta1.AppliedTo, ingress, egress []crdv1beta1.Rule) []string {
+	appliedToNode := false
+	for _, eachAppliedTo := range specAppliedTo {
+		if eachAppliedTo.NodeSelector != nil {
+			appliedToNode = true
+			break
+		}
+	}
+	if !appliedToNode {
+		return nil
+	}
+	var warnings []string
+	for _, eachIngress := range ingress {
+		if eachIngress.EnableLogging && len(eachIngress.LogLabel) > 12 {
+			warnings = append(warnings, fmt.Sprintf(`Policy applied to Node rule "%s" logLabel "%s" is too long. LogLabel over 12 characters will be truncated; please use 12 or fewer characters`, eachIngress.Name, eachIngress.LogLabel))
```

Comment:
```suggestion
			warnings = append(warnings, fmt.Sprintf("LogLabels for Node NetworkPolicies are limited to 12 characters, but label %q for policy rule %q exceeds the limit and will be truncated in kernel logs", eachIngress.LogLabel, eachIngress.Name))
```

ditto for egress below

Reply:
Done